### PR TITLE
Hooks and notifications support for libplugin

### PR DESCRIPTION
**Pull request by darosior**

### CODE REVIEW

File: plugins/libplugin.h [hunk 1]
```diff
@@ -50,6 +50,14 @@ struct plugin_notification {
...
 };
 
+/* Create an array of these, one for each hook you subscribe to. */
+struct plugin_hook {
+	const char *name;
+	struct command_result (*handle)(struct command *cmd,
+	                                const char *buf,
+	                                const jsmntok_t *params);
+};
+
```
Comment (Saibato) [inline plugins/libplugin.h]: LGTM tested notifications and hooks. :+1: 
I had to change  

> struct command_result (*handle)(struct command *cmd,
	                                const char *buf,
	                                const jsmntok_t *params);
}

to a function pointer to command_result

To get the hook to return a command_success(cmd, ...);


```
diff --git a/plugins/libplugin.h b/plugins/libplugin.h
index d79e7e53..b1df9cd7 100644
--- a/plugins/libplugin.h
+++ b/plugins/libplugin.h
@@ -53,7 +53,7 @@ struct plugin_notification {
 /* Create an array of these, one for each hook you subscribe to. */
 struct plugin_hook {
        const char *name;
-       struct command_result (*handle)(struct command *cmd,
+       struct command_result *(*handle)(struct command *cmd,
                                        const char *buf,
                                        const jsmntok_t *params);
 };

```

Comment (darosior) [reply plugins/libplugin.h]: Oops, thanks !

**Comment by Saibato** [comment]
tACK https://github.com/ElementsProject/lightning/pull/3317/commits/7a0d22508e5eb77026ee5c547886b3f1322ac276 so now they can write full plugs without using RD languages and its dependencies :heart_eyes: 

<details><summary>since i see non c/asm/bytecode coding as waste of time and resources, here to boost adoption some simple testcode + makefile changes</summary>

https://github.com/ElementsProject/lightning/commit/83a66e26ad1a50d125c0e9ffb2589e53b8a3be37
</details>

**Comment by darosior** [comment]
> so now they can write full plugs without using RD languages and its dependencies heart_eyes

Yeah I'm on it for bitcoin-cli interaction ;-)

**Comment by cdecker** [comment]
Excellent addition @darosior :+1: Thanks for the reviews @Saibato ^^

ACK 7a0d22508e5eb77026ee5c547886b3f1322ac276

**Pull request merged**